### PR TITLE
feat: Copy Backup

### PR DESCRIPTION
**Pull request by Mariatta**

### CODE REVIEW

File: google/cloud/bigtable/backup.py [hunk 1]
```diff
@@ -339,6 +356,68 @@ def create(self, cluster_id=None):
...
+        cluster_id=None,
+        expire_time=None,
+    ):
+        """Make a copy of this backup.
+
+        :type new_backup_id: str
+        :param new_backup_id: The name of the copied backup
+
+        :type project_id: str
+        :param project_id: The destination project id for the backup
```
Comment (kolea2) [inline google/cloud/bigtable/backup.py:373]: is this also optional? same question for other parameters

File: google/cloud/bigtable/backup.py [hunk 2]
```diff
@@ -339,6 +356,68 @@ def create(self, cluster_id=None):
...
+
+        parent = BigtableTableAdminClient.cluster_path(
+            project=project_id,
+            instance=instance_id,
+            cluster=cluster_id,
+        )
+
+        return api.copy_backup(
+            request={
+                "parent": parent,
```
Comment (kolea2) [inline google/cloud/bigtable/backup.py:414]: how does this work if the source backup project and parent backup project are different?

File: tests/system/test_table_admin.py [hunk 3]
```diff
@@ -323,6 +324,27 @@ def test_table_backup(
         temp_table_backup.expire_time
     )
 
+    # Testing `backup.copy()` method
+    backup_op = temp_backup.copy("copied-backup")
```
Comment (kolea2) [inline tests/system/test_table_admin.py:328]: I would also add some test cases with some of the other parameters set.

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: google/cloud/bigtable/backup.py (hunk 2 above)
Comment (Mariatta) [reply google/cloud/bigtable/backup.py:414]: `parent` is constructed using `project_id`, `instance_id`, `cluster_id`. It's allowed to be different I believe as long as the user have access to both. I'm not seeing any restriction about it from the design doc. From the client we will pass the path, and relay back error from the server.

File: tests/system/test_table_admin.py (hunk 3 above)
Comment (Mariatta) [reply tests/system/test_table_admin.py]: I've added tests that pass additional parameters.

File: tests/system/test_table_admin.py [hunk 4]
```diff
@@ -354,3 +350,103 @@ def test_table_backup(
...
+    temp_backup = source_table.backup(
+        temp_backup_id,
+        cluster_id=data_cluster_id,
+        expire_time=expire,
+    )
+    temp_backup.create()
+    backups_to_delete.append(temp_backup)
+
+    # Test copying to the same instance
+    backup_op = temp_backup.copy("copied-backup")
```
Comment (kolea2) [inline tests/system/test_table_admin.py:393]: let's add a unique identifier to `copied-backup`

File: tests/system/test_table_admin.py [hunk 5]
```diff
@@ -354,3 +350,103 @@ def test_table_backup(
...
+    alt_cluster = alt_instance.cluster(
+        cluster_id=alt_cluster_id,
+        location_id=location_id,
+        serve_nodes=1,
+    )
+    alt_instance.create(clusters=[alt_cluster])
+    instances_to_delete.append(alt_instance)
+
+    # create a copy to the new instance, cluster, and new expire_time
+    copied_backup_id = "copy-new-instance"
```
Comment (kolea2) [inline tests/system/test_table_admin.py:424]: same here

File: tests/system/test_table_admin.py (hunk 4 above)
Comment (Mariatta) [reply tests/system/test_table_admin.py]: I've added `-{unique_suffix}` to the copy backup ids

**Pull request closed without merging**